core/bin/snapshots_creator/src/main.rs

```diff
@@ -291,5 +295,10 @@ async fn main() -> anyhow::Result<()> {
     run(blob_store, replica_pool, master_pool, MIN_CHUNK_COUNT).await?;
     tracing::info!("Finished running snapshot creator!");
     stop_sender.send(true).ok();
+    if let Some(prometheus_exporter_task) = prometheus_exporter_task {
+        prometheus_exporter_task
```

Comment:
Nit: It may make sense to have a reasonable timeout here (e.g., 10s) via `tokio::time::timeout` so that it's guaranteed that the process eventually terminates. Maybe, that's handled by K8s infra already; in any case, feel free to leave as-is if you don't think it's ebtter.

Reply:
AFAIK We have alerts if a job is running too long, so we'd know if this ever happens